numbat/examples/inspect.rs

```diff
@@ -86,25 +79,137 @@ fn inspect_functions_in_module(ctx: &Context, module: String) {
+                let extra_import = if !example_ctx
+                    .resolver()
+                    .imported_modules
+                    .contains(&module_path)
+                {
+                    format!("use {}\n", module)
+                } else {
+                    "".into()
+                };
+                let _result = example_ctx
+                    .interpret(&extra_import, CodeSource::Internal)
+                    .unwrap();
+
+                if let Ok((statements, results)) =
+                    example_ctx.interpret(&example_code, CodeSource::Internal)
+                {
+                    //Format the example input
+                    let example_input = format!(">>> {}", example_code);
+
+                    //Encode the example url
+                    let url_code = extra_import + &example_code;
+                    let example_url = format!(
+                        "https://numbat.dev/?q={}",
+                        percent_encoding::utf8_percent_encode(
+                            &url_code,
+                            percent_encoding::NON_ALPHANUMERIC
+                        )
+                    );
+
+                    //Assemble the example output
+                    let mut example_output = String::new();
+                    example_output += "\n";
+
+                    for statement in &statements {
+                        example_output += &plain_text_format(&statement.pretty_print(), true);
+                        example_output += "\n\n";
+                    }
+
+                    let result_markup = results.to_markup(
+                        statements.last(),
+                        &example_ctx.dimension_registry(),
+                        true,
+                        true,
+                    );
+                    example_output += &plain_text_format(&result_markup, false);
+
+                    if results.is_value() {
+                        example_output += "\n";
+                    }
+
+                    //Print the example
+                    if let Some(example_description) = example_description {
+                        println!(
+                        "* {}\n\n  <a href=\"{}\"><i class=\"fa fa-play\"></i> Run this example</a>",
+                        replace_equation_delimiters(example_description),
+                        example_url
+                    );
+                    } else {
+                        println!(
+                            "* <a href=\"{}\"><i class=\"fa fa-play\"></i> Run this example</a>\n",
+                            example_url
+                        );
+                    }
+
+                    println!("  ```nbt");
+                    for l in example_input.lines() {
+                        println!("    {}", l);
+                    }
+                    for l in example_output.lines() {
+                        println!("    {}", l);
+                    }
+                    println!("  ```");
```

Comment:
Can we trim the final newline in the examples? That would make the output a bit more concise.

![image](https://github.com/user-attachments/assets/52e901fa-c894-4632-a07e-35b82ed1b970)

Reply:
Yes or course, I initially tried to get the output exactly as it would appear if the input was typed in an interactive session where there is the extra empty line at the end but it does make sense to save some space.